src/software/ai/hl/stp/play/defense/defense_play_fsm.cpp

```diff
@@ -0,0 +1,142 @@
+    std::vector<DefenderAssignment> crease_defender_assignments;
+    std::vector<DefenderAssignment> pass_defender_assignments;
+    for (unsigned int i = 0; i < event.common.num_tactics; i++)
+    {
+        DefenderAssignment defender_assignment;
+        if (i < assignments.size())
+        {
+            defender_assignment = assignments.at(i);
+        }
+        else
+        {
+            // If we have more tactics to set than determined defender assignments,
+            // assign remaining defenders to the defender assignment with the
+            // highest coverage rating
+            defender_assignment = assignments.front();
+        }
+
+        if (defender_assignment.type == CREASE_DEFENDER)
+        {
+            crease_defender_assignments.emplace_back(defender_assignment);
+
+            // If we have at least two available defenders, two defenders should
+            // be assigned to the highest scoring crease defender assignment to better
+            // block the shot cone of the most threatening enemy
+            if (i == 0 && event.common.num_tactics >= 2)
+            {
+                crease_defender_assignments.emplace_back(defender_assignment);
+                i++;
+            }
+        }
+        else
+        {
+            pass_defender_assignments.emplace_back(defender_assignment);
+        }
+    }
+
+    // Reset tactics if the number of crease defenders or pass defenders
+    // we intend to assign has changed
+    setUpCreaseDefenders(static_cast<unsigned int>(crease_defender_assignments.size()));
+    setUpPassDefenders(static_cast<unsigned int>(pass_defender_assignments.size()));
+
+    for (unsigned int i = 0; i < crease_defenders.size(); i++)
+    {
+        auto target = crease_defender_assignments.at(i).target;
+
+        // Determine the number of crease defenders already assigned to the target
+        auto defenders_with_target_count = std::count_if(
+            crease_defender_assignments.begin(), crease_defender_assignments.begin() + i,
+            [&target](const auto& assignment) { return assignment.target == target; });
+
+        // Pick alignment based on how many crease defenders are already assigned to the
+        // target
+        TbotsProto::CreaseDefenderAlignment alignment;
+        switch (defenders_with_target_count)
+        {
+            case 0:
+                alignment = TbotsProto::CreaseDefenderAlignment::CENTRE;
+                break;
+            case 1:
+                alignment = (event.common.world.ball().position().y() > 0)
+                                ? TbotsProto::CreaseDefenderAlignment::LEFT
+                                : TbotsProto::CreaseDefenderAlignment::RIGHT;
+                break;
+            case 2:
+                alignment = (event.common.world.ball().position().y() > 0)
+                                ? TbotsProto::CreaseDefenderAlignment::RIGHT
+                                : TbotsProto::CreaseDefenderAlignment::LEFT;
```

Comment:
this style of assignment is not preferred

Reply:
done